frontend_app_file/src/component/Kanban.jsx

```diff
@@ -0,0 +1,386 @@
+            break
+        }
+        break
+      default:
+        alert(props.t('Error while saving the new version'))
+        break
+    }
+  }
+
+  handleAddColumnClicked = (board, column) => {
+    // console.log("handleAddColumnClicked", board, column)
+    const title = prompt('Please enter the name of the new column')
+    if (!title) {
+      return
+    }
+
+    this.handleNewColumnConfirm({
+      title: title,
+      cards: []
+    })
+  }
+
+  handleCardRemove = (card) => {
+    // console.log("handleCardRemove", card)
+    this.removeCard(card)
+  }
+
+  handleColumnRemove = (column) => {
+    // console.log("handleColumnRemove", column)
+    this.updateColumns(this.state.board.columns.filter(col => col !== column))
+  }
+
+  handleCardDragEnd = (card, { fromColumnId, fromPosition }, { toColumnId, toPosition }) => {
+    // console.log("handleCardDragEnd", card, { fromColumnId, fromPosition }, {toColumnId, toPosition })
+    this.updateColumns(
+      this.state.board.columns.map(
+        col => (
+          (col.id === fromColumnId && col.id === toColumnId)
+            ? { ...col, cards: moveElem(col.cards, fromPosition, toPosition, card) }
+            : (
+              col.id === fromColumnId
+                ? { ...col, cards: col.cards.filter(c => c !== card) }
+                : (
+                  col.id === toColumnId
+                    ? {
+                      ...col,
+                      cards: [
+                        ...col.cards.slice(0, toPosition),
+                        card,
+                        ...col.cards.slice(toPosition)
+                      ]
+                    }
+                    : col
+                )
+            )
+        )
+      )
+    )
+    //     this.updateBoard(board)
+  }
+
+  handleNewColumnConfirm = (column) => {
+    // console.log("handleNewColumnConfirm", column)
+    const newColumn = { ...column, id: uuidv4() }
+
+    this.updateColumns([
+      ...this.state.board.columns,
+      newColumn
+    ])
+  }
+
+  handleColumnDragEnd = (column, { fromPosition }, { toPosition }) => {
+    this.updateColumns(
+      moveElem(
+        this.state.board.columns,
+        fromPosition,
+        toPosition,
+        column
+      )
+    )
+  }
+
+  updateColumns (newColumns) {
+    this.updateBoard({ ...this.state.board, columns: newColumns })
+  }
+
+  updateBoard (newBoard, dontSaveDraftToLocalStorage) {
+    this.setState({
+      board: newBoard,
+      mustSave: !dontSaveDraftToLocalStorage
+    })
+
+    if (!dontSaveDraftToLocalStorage) {
+      setLocalStorageItem('rawContent', this.props.content, 'kanban', JSON.stringify(newBoard))
+    }
+  }
+
+  handleColumnRenameClick = (column) => {
+    const newName = prompt(this.props.t('Please enter the new name of the column'), column.title)
+    if (newName) {
+      this.handleColumnRename(column, newName)
+    }
+  }
+
+  handleColumnRename = (column, title) => {
```

Comment:
"handlers" are for user's action, I wouldn't call this function with the word "handle"

Reply:
This is called back by the kanban component as after a user action

Reply:
then it's not a handler, its more of a utility function